src/IMG_avif.c

```diff
@@ -756,3 +757,785 @@ bool IMG_SaveAVIF_IO(SDL_Surface *surface, SDL_IOStream *dst, bool closeio, int
+                    SDL_SetError("Couldn't lock AVIF frame surface for writing");
+                    error = true;
+                    goto done;
+                }
+                lockedSurf = true;
+            }
+
+            rgb.pixels = (uint8_t *)frame_surface->pixels;
+            rgb.rowBytes = (uint32_t)frame_surface->pitch;
+            result = lib.avifImageYUVToRGB(image, &rgb);
+
+            if (lockedSurf) {
+                SDL_UnlockSurface(frame_surface);
+            }
+
+            if (result != AVIF_RESULT_OK) {
+                SDL_SetError("Couldn't convert AVIF image to RGB: %s", lib.avifResultToString(result));
+                SDL_DestroySurface(frame_surface);
+                frame_surface = NULL;
+                error = true;
+                goto done;
+            }
+
+            SDL_Colorspace colorspace = SDL_DEFINE_COLORSPACE(SDL_COLOR_TYPE_RGB,
+                                                              SDL_COLOR_RANGE_FULL,
+                                                              image->colorPrimaries,
+                                                              image->transferCharacteristics,
+                                                              image->matrixCoefficients,
+                                                              SDL_CHROMA_LOCATION_NONE);
+            SDL_SetSurfaceColorspace(frame_surface, colorspace);
+        }
+
+        animation->frames[i] = frame_surface;
+        frame_surface = NULL;
+
+        if (decoder->imageTiming.timescale > 0) {
+            frame_duration_ms = (double)decoder->imageTiming.durationInTimescales * 1000.0 / decoder->imageTiming.timescale;
+            animation->delays[i] = (int)SDL_round(frame_duration_ms);
+        } else {
+            animation->delays[i] = 0;
+        }
+
+        if (i == 0) {
+            animation->w = animation->frames[0]->w;
+            animation->h = animation->frames[0]->h;
+        }
+    }
+
+done:
+    if (decoder) {
+        lib.avifDecoderDestroy(decoder);
+    }
+    if (frame_surface) {
+        SDL_DestroySurface(frame_surface);
+    }
+    if (error) {
+        if (animation) {
+            if (animation->frames) {
+                for (i = 0; i < animation->count; ++i) {
+                    if (animation->frames[i]) {
+                        SDL_DestroySurface(animation->frames[i]);
+                    }
+                }
+                SDL_free(animation->frames);
+            }
+            if (animation->delays) {
+                SDL_free(animation->delays);
+            }
+            SDL_free(animation);
+        }
+        SDL_SeekIO(src, start, SDL_IO_SEEK_SET);
+        return NULL;
+    }
+    return animation;
+}
+#else
+
+IMG_Animation* IMG_LoadAVIFAnimation_IO(SDL_IOStream* src)
+{
+    (void)src;
+    SDL_SetError("SDL_image built without AVIF animation loading support");
+    return NULL;
+}
+
+#endif /* LOAD_AVIF */
+
+#if SDL_IMAGE_SAVE_AVIF
```

Comment:
```suggestion
#if SAVE_AVIF
```
The other formats don't use the `SDL_IMAGE_` prefix. We could add that as a separate commit if we really wanted.